File: example/vegas/vegas2.jl
using MCIntegration

result = integrate(neval=10000, dof=[[1], [1], [1], [1]], print=0) do c
    X = c.var[1]
    dx2 = 0.0
    for d in 1:4
        dx2 += (X[d] - 0.5)^2
    end
    f = exp(-200 * dx2)
    # f = exp(-1 * dx2)
    if c.curr == 1
        return f
    elseif c.curr == 2
        return f * (x[1] + 1e-4)
    else
        return f * (x[1] + 1e-4)^2
    end
end

MCIntegration.summary(result, [obs -> obs[1], obs -> obs[2], obs -> obs[3]]; verbose=-1)

File: example/vegas/vegas2.py
import vegas
import math
import gvar as gv


def f(x):
    dx2 = 0.0
    for d in range(4):
        dx2 += (x[d] - 0.5) ** 2
    f = math.exp(-200 * dx2)
    return [f, f * x[0], f * x[0] ** 2]


integ = vegas.Integrator(4 * [[0, 1]])

# adapt grid
training = integ(f, nitn=10, neval=2000)

# final analysis
result = integ(f, nitn=10, neval=10000)
print('I[0] =', result[0], '  I[1] =', result[1], '  I[2] =', result[2])
print('Q = %.2f\n' % result.Q)
print('<x> =', result[1] / result[0])
print(
    'sigma_x**2 = <x**2> - <x>**2 =',
    result[2] / result[0] - (result[1] / result[0]) ** 2
)
print('\ncorrelation matrix:\n', gv.evalcorr(result))
